dev/src/main/java/org/eclipse/codewind/intellij/ui/form/AppOverviewFrame.java

```diff
@@ -0,0 +1,322 @@
+    private JLabel containerIdLabel;
+    private JTextArea containerIdTextArea;
+    private JLabel statusLabel;
+    private JTextField statusField;
+    private JLabel applicationContextRootLabel;
+    private JTextField applicationContextRootField;
+    private JLabel internalApplicationPortLabel;
+    private JTextField internalApplicationPortField;
+    private JLabel internalDebugPortLabel;
+    private JTextField internalDebugPortField;
+    private JButton editProjectSettingsButton;
+    private JButton moreInformationButton;
+    private JPanel buttonPanel;
+    private JLabel autoBuildLabel;
+    private JTextField autoBuildField;
+    private JLabel autoMetricsInjectionLabel;
+    private JTextField autoMetricsInjectionField;
+    private JLabel lastBuildLabel;
+    private JFormattedTextField lastBuildField;
+    private JLabel lastImageBuildLabel;
+    private JTextField lastImageBuildField;
+    private JSeparator separatorSpace;
+    private JPanel footerPanel;
+    private JPanel horizontalFiller;
+    private JButton refreshButton;
+    private JLabel preferencesLink;
+
+    private CodewindApplication application;
+    private final Project project;
+
+    public AppOverviewFrame(CodewindApplication application, Project project) {
+        this.application = application;
+        this.project = project;
+    }
+
+    private void createUIComponents() {
+        final CodewindApplication app = this.application;
+        final Project proj = this.project;
+        overviewPanel = new JPanel();
+        overviewPanel.setBackground(JBColor.background());
+        // Top section showing app name
+        appLabel = new JLabel(this.application.name);
+        appLabel.setIcon(IconCache.getCachedIcon(IconCache.ICONS_CODEWIND_PNG));
+        // General Pane
+        typeField = new JTextField(); // Type
+        languageField = new JTextField(); // Language
+        locationOnDiskTextArea = WidgetUtils.createJTextArea(""); // Location on Disk
+        applicationUrlLink = WidgetUtils.createHyperlink(""); // Application Endpoint
+        externalApplicationPortField = new JTextField(); // External/Exposed Application Port
+        externalDebugPortField = new JTextField(); // External/Exposed Debug Port
+        projectIdTextArea = WidgetUtils.createJTextArea(""); // Project ID
+        containerIdTextArea = WidgetUtils.createJTextArea(""); // Container ID
+        statusField = new JTextField(); // Status
+
+        // Project Settings Pane
+        applicationContextRootField = new JTextField(); // Application Context Root
+        internalApplicationPortField = new JTextField(); // Internal Application Port
+        internalDebugPortField = new JTextField(); // Internal Debug Port
+
+        editProjectSettingsButton = new JButton();
+        editProjectSettingsButton.setText(message("AppOverviewEditorEditProjectSettings"));
+        editProjectSettingsButton.setToolTipText("Edit Codewind settings file"); // Todo: Externalize
```

Comment:
We should do the todo and externalize this now.

Reply:
Done